### PR TITLE
image-tests: don't use random names for CI artifacts

### PR DESCRIPTION
**Pull request by msehnout**
When using random names for artifacts like AWS snapshots, or Azure
images, it becomes hard to clean them up in case of CI failure. See this
issue for more details:
https://github.com/osbuild/osbuild-composer/issues/942

This PR introduces predictable names so that we can easily determine
which artifact belongs to which PR and therefore we can decide to wipe
all resources that are not needed any more.

### CODE REVIEW

File: internal/boot/helpers.go
Comment (larskarlitski) [inline internal/boot/helpers.go:73]: I don't like reading environment variables from a library, because it makes it hard to figure out how a program is meant to be run. Is there a way to pass those through from `main_test.go`?

Don't block on it, especially since there seems to be a lot of such use here, and it's only test code.

File: schutzbot/run_image_tests.sh
Comment (larskarlitski) [inline schutzbot/run_image_tests.sh:65]: Is this necessary because of the additional environment variables that are read above? Why wasn't it necessary for the other vars?

**Review by larskarlitski — CHANGES_REQUESTED** [review]
I really like this, thanks!

Please drop `GenerateRandomStrings` now. It doesn't seem to be used anymore.

File: internal/boot/helpers.go
Comment (msehnout) [reply internal/boot/helpers.go:73]: We could move the whole function out of the library into `main_test.go`, the only function which requires separate solutions is `WithBootedImageInEC2` because it also creates a security group, but we can pass the change/build id to it.

File: schutzbot/run_image_tests.sh
Comment (larskarlitski) [reply schutzbot/run_image_tests.sh:65]: Ah I see. Thanks!

**Comment by msehnout** [comment]
@larskarlitski the PR is again ready for review. RHEL8.3 stages fail due to corrupted mock repo and I don't know what is wrong with codecov, but I can't see any errors in the UI.

**Review by larskarlitski — APPROVED** [review]
> @larskarlitski the PR is again ready for review. RHEL8.3 stages fail due to corrupted mock repo and I don't know what is wrong with codecov, but I can't see any errors in the UI.

Nice! Thanks.

The corrupt repo error seems to be gone from the latest log, but vhd-boot is failing. Is that unrelated?

I've restarted codecov. Seems to have failed running.

**Comment by msehnout** [comment]
> The corrupt repo error seems to be gone from the latest log, but vhd-boot is failing. Is that unrelated?

This seems related, I'll check it.

File: schutzbot/Jenkinsfile
Comment (ondrejbudai) [inline schutzbot/Jenkinsfile:136]: I think you can reuse `ID` and `VERSION_ID`. See https://github.com/osbuild/osbuild-composer/blob/85423d403c031f9931568bb2525bcb2d4dd7eb94/test/image-tests/aws.sh#L7

**Review by ondrejbudai — COMMENTED** [review]
Looks very nice, thanks! See my note inline.

File: schutzbot/Jenkinsfile
Comment (msehnout) [reply schutzbot/Jenkinsfile:136]: I was afraid that it would clash for RHEL 8 in case we test against beta and nightly channels. This seemed to be a simple and safe workaround, but if you think there is nothing to worry about I can change it for sure.

**Comment by msehnout** [comment]
> > The corrupt repo error seems to be gone from the latest log, but vhd-boot is failing. Is that unrelated?
> 
> This seems related, I'll check it.

I haven't found anything that could cause this yet. So I at least tried restarting the job. It passed this time, but failed with VMWare this time :(

File: schutzbot/Jenkinsfile
Comment (ondrejbudai) [reply schutzbot/Jenkinsfile:136]: Right, let's merge this and worry about this later if we want to.

**Review by ondrejbudai — APPROVED** [review]
Very nice, thanks!

**Comment by ondrejbudai** [comment]
For the record, VMWare was broken yesterday, this was an unrelated failure.

**Pull request merged**
